### PR TITLE
#169258619: Configure ESLint in the project

### PR DESCRIPTION
**Pull request by nziokaivy**
#### What does this PR do?
Configure ESLint in the project.

#### Description of Task to be completed?
- Install eslint
- Configure eslint 
- Add eslint rules in the configuration file

#### How should this be manually tested?
- Clone this repository 
- Run the command `npm run eslint`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169258619](https://www.pivotaltracker.com/story/show/169258619)

#### Screenshots (if appropriate)

N/A
#### Questions:

### CODE REVIEW

**Review by victkarangwa — APPROVED** [review]

**Review by WillyBoris — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      }],
+      "require-jsdoc": ["error", {
+          "require": {
+              "FunctionDeclaration": true,
+              "MethodDefinition": true,
+              "ClassDeclaration": true
+          }
+      }]
+    }
+  }
```
Comment (julietezekwe) [inline .eslintrc.json]: End of line

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      "consistent-return": 0,
+      "no-param-reassign": 0,
+      "comma-dangle": 0,
+      "curly": ["error", "multi-line"],
+      "import/no-unresolved": [2, { "commonjs": true }],
+      "no-shadow": ["error", { "allow": ["req", "res", "err"] }],
+      "valid-jsdoc": ["error", {
+        "requireReturn": true,
+        "requireReturnType": true,
+        "requireParamDescription": false,
```
Comment (julietezekwe) [inline .eslintrc.json]: Any reason why `requireParamDescription` is false?

File: package.json [hunk 3]
```diff
@@ -14,6 +15,8 @@
     "dotenv": "^6.0.0",
     "ejs": "^2.6.1",
     "errorhandler": "^1.5.0",
+    "eslint-config-airbnb-base": "^14.0.0",
+    "eslint-plugin-react": "^7.16.0",
```
Comment (hezronkimutai) [inline package.json]: Nice work Ivy, but I don't understand why we need eslint-plugin-react

File: package.json [hunk 4]
```diff
@@ -14,6 +15,8 @@
     "dotenv": "^6.0.0",
     "ejs": "^2.6.1",
     "errorhandler": "^1.5.0",
+    "eslint-config-airbnb-base": "^14.0.0",
```
Comment (julietezekwe) [inline package.json]: Move to devDependency

File: package.json (hunk 3 above)
Comment (julietezekwe) [inline package.json]: Move to devDependency

**Review by julietezekwe — CHANGES_REQUESTED** [review]
Great work @nziokaivy. Kindly attend to the comments.

File: package.json (hunk 3 above)
Comment (nziokaivy) [reply package.json]: @hezronkimutai I figured we'd better have it now since we will be using react for the frontend?

Comment (julietezekwe) [reply package.json]: Kindly resolve this comments @nziokaivy

File: .eslintrc.json (hunk 2 above)
Comment (nziokaivy) [reply .eslintrc.json]: Oh Sorry! meant for it to be true

File: package.json (hunk 3 above)
Comment (hezronkimutai) [reply package.json]: @nziokaivy  but the front end will be on another repo.

**Review by NiyoEric — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -26,5 +27,9 @@
     "passport-local": "^1.0.0",
     "request": "^2.87.0",
     "underscore": "^1.9.1"
+  },
+  "devDependencies": {
```
Comment (nziokaivy) [inline package.json:31]: @hezronkimutai Please view that I have removed the eslint react dependency.

**Review by julietezekwe — COMMENTED** [review]
I see you have fixed the issues, Please make your commits one and rebase from develop to have the latest work on your branch then reach out to me.

**Review by julietezekwe — APPROVED** [review]

**Pull request merged**